### PR TITLE
[Glimmer2] Remove `{{#view}}` helper tests

### PR DESCRIPTION
**Pull request by chadhietala**
This currently removes just the tests related to the `view` helper. I have left some tests in the file because I have a feeling that these tests are cross cutting and were used as coverage for components. I could be 100% incorrect about this but would like to hear from @rwjblue and @chancancode before they are removed :bomb:.

Furthermore, it looks like we will need to dereference all the `view` helpers in all of the tests before we can finally remove the helper. Luckily majority of the tests in #13127 should be addressing this.

Part of #13127

### CODE REVIEW

**Comment by chadhietala** [comment]
Going to follow up with another PR to add some tests to `curly-component` tests that should cover these types of test cases.

File: packages/ember-htmlbars/tests/helpers/view_test.js [hunk 1]
```diff
@@ -366,436 +61,99 @@ QUnit.test('Should apply class without condition always', function() {
...
-  owner.register('template:foo', compile('{{#view view.idView class="bar"}}baz{{/view}}'));
+  run(function() {
+    set(view, 'output', new SafeString('you are so <i>super</i>'));
+  });
 
-  var IdView = EmberView;
+  equal(view.$('i').length, 1, 'creates an element when value is updated');
+});
 
+QUnit.test('should teardown observers from bound properties on rerender', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/view_test.js]: This appears to be covered in https://github.com/emberjs/ember.js/blob/master/packages/ember-views/tests/views/view/attribute_bindings_test.js#L256

File: packages/ember-htmlbars/tests/helpers/view_test.js [hunk 2]
```diff
@@ -807,20 +165,6 @@ QUnit.test('{{view}} should not allow attributeBindings to be set', function() {
...
-      template: compile('common')
-    })
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), 'common', 'tries to look up view name locally');
-});
-
 QUnit.test('{{view}} should evaluate class bindings set in the current context', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/view_test.js]: This is `{{#view}}` specific.

File: packages/ember-htmlbars/tests/helpers/view_test.js [hunk 3]
```diff
@@ -850,18 +194,6 @@ QUnit.test('{{view}} should evaluate class bindings set in the current context',
...
-    textField: TextField,
-    template: compile('{{view view.textField value=view.name}}')
-  });
-
-  runAppend(view);
-
-  equal(view.$('input').val(), 'myView', 'evaluates attributes bound in the current context');
-});
-
 QUnit.test('{{view}} should be able to bind class names to truthy properties', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/view_test.js]: This is `{{#view}}` specific.

File: packages/ember-htmlbars/tests/helpers/view_test.js [hunk 4]
```diff
@@ -366,436 +61,99 @@ QUnit.test('Should apply class without condition always', function() {
...
-  runAppend(view);
-
-  var childView = get(view, 'childViews.firstObject');
-  equal(view.$().children().first().children().first().attr('id'), get(childView, 'elementId'));
-});
-
-QUnit.test('Child views created using the view helper should have their parent view set properly', function() {
-  var template = '{{#view}}{{#view}}{{view}}{{/view}}{{/view}}';
-
+QUnit.test('Should not apply classes when bound property specified is false', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/view_test.js]: Now covered in https://github.com/emberjs/ember.js/pull/13236/files#diff-c47395a5aaf92aa0a7edbfa812d4e0afR108

File: packages/ember-htmlbars/tests/helpers/view_test.js [hunk 5]
```diff
@@ -366,436 +61,99 @@ QUnit.test('Should apply class without condition always', function() {
...
-  id = childView.$()[0].id;
-  equal(id, 'templateViewTest', 'precond -- id of childView should be set correctly');
-  equal(EmberView.views[id], childView, 'childView with passed ID is registered with View.views so that it can properly receive events from EventDispatcher');
+  ok(jQuery('#foo').hasClass('some-prop'), 'adds dasherized class when value is true');
 });
 
-QUnit.test('Child views created using the view helper and that have a viewName should be registered as properties on their parentView', function() {
-  var template = '{{#view}}{{view viewName="ohai"}}{{/view}}';
-
+QUnit.test('should escape HTML in normal mustaches', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/view_test.js]: Now covered in https://github.com/emberjs/ember.js/pull/13236/files#diff-c47395a5aaf92aa0a7edbfa812d4e0afR660

File: packages/ember-htmlbars/tests/helpers/view_test.js [hunk 6]
```diff
@@ -366,436 +61,99 @@ QUnit.test('Should apply class without condition always', function() {
...
+  equal(view.$().text(), 'you are so <i>super</i>', 'updates with entities, not elements');
+  equal(view.$('i').length, 0, 'does not create an element when value is updated');
 });
 
-QUnit.test('{{view}} id attribute should set id on layer', function() {
-  owner.register('template:foo', compile('{{#view view.idView id="bar"}}baz{{/view}}'));
-
-  var IdView = EmberView;
-
+QUnit.test('should not escape HTML in triple mustaches', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/view_test.js]: Now covered in https://github.com/emberjs/ember.js/pull/13236/files#diff-c47395a5aaf92aa0a7edbfa812d4e0afR687

File: packages/ember-htmlbars/tests/helpers/view_test.js [hunk 7]
```diff
@@ -366,436 +61,99 @@ QUnit.test('Should apply class without condition always', function() {
...
-  owner.register('template:foo', compile('{{#view view.tagView tag="span"}}baz{{/view}}'));
+  run(function() {
+    set(view, 'output', 'you are so <i>super</i>');
+  });
 
-  var TagView = EmberView;
+  equal(view.$('i').length, 1, 'creates an element when value is updated');
+});
 
+QUnit.test('should not escape HTML if string is a Handlebars.SafeString', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/view_test.js]: Now covered in https://github.com/emberjs/ember.js/pull/13236/files#diff-c47395a5aaf92aa0a7edbfa812d4e0afR724

File: packages/ember-htmlbars/tests/helpers/view_test.js [hunk 8]
```diff
@@ -366,436 +61,99 @@ QUnit.test('Should apply class without condition always', function() {
...
+  equal(observersFor(view, 'foo').length, 1);
+
+  run(function() {
+    view.rerender();
+  });
+
+  equal(observersFor(view, 'foo').length, 1);
 });
 
 QUnit.test('{{view}} should not allow attributeBindings to be set', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/view_test.js]: Now covered in https://github.com/emberjs/ember.js/pull/13236/files#diff-c47395a5aaf92aa0a7edbfa812d4e0afR373

File: packages/ember-htmlbars/tests/helpers/view_test.js [hunk 9]
```diff
@@ -910,305 +242,6 @@ QUnit.test('{{view}} should be able to bind class names to truthy or falsy prope
...
-      })
-    })
-  });
-
-  expectAssertion(function() {
-    runAppend(view);
-  }, /must be a subclass or an instance of Ember.View/);
-});
-
 QUnit.test('Specifying `id` to {{view}} is set on the view.', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/view_test.js]: Now covered in https://github.com/emberjs/ember.js/pull/13236/files#diff-c47395a5aaf92aa0a7edbfa812d4e0afR23

**Comment by chadhietala** [comment]
Will delete all the tests once I get a thumbs up.

File: packages/ember-htmlbars/tests/helpers/view_test.js [hunk 10]
```diff
@@ -366,436 +61,99 @@ QUnit.test('Should apply class without condition always', function() {
...
 
-  var childView = firstGrandchild(view);
-  equal(childView, get(firstChild(childView), 'parentView'), 'parent view is correct');
+  ok(!jQuery('#foo').hasClass('some-prop'), 'does not add class when value is falsey');
 });
 
-QUnit.test('Child views created using the view helper should have their IDs registered for events', function() {
-  var template = '{{view}}{{view id="templateViewTest"}}';
-
+QUnit.test('Should apply classes of the dasherized property name when bound property specified is true', function() {
```
Comment (chadhietala) [inline packages/ember-htmlbars/tests/helpers/view_test.js]: Appears to be `{{#view}}` specific.

**Comment by rwjblue** [comment]
Merged https://github.com/emberjs/ember.js/pull/13236, does this need to be updated?

File: packages/ember-template-compiler/lib/index.js [hunk 11]
```diff
@@ -35,7 +34,6 @@ registerPlugin('ast', AssertNoEachIn);
 
 if (!_Ember.ENV._ENABLE_LEGACY_VIEW_SUPPORT) {
   registerPlugin('ast', AssertNoViewAndControllerPaths);
-  registerPlugin('ast', AssertNoViewHelper);
```
Comment (rwjblue) [inline packages/ember-template-compiler/lib/index.js]: This likely should stick around (moved outside the `if`) to provide feedback to folks that might try to use `{{view 'foo'}}` after `{{view` keyword was removed.

**Comment by chadhietala** [comment]
@rwjblue Not sure how people feel about deleting the tests but keep the helper around. I was thinking we would wait till the `{{#view}}` is no longer used internally for tests before merging this. If people are ok with loosing the coverage for `{{#view}}` for the time being then I can address the transform issue and then this should be in a reviewable state.

**Comment by chadhietala** [comment]
There are pending PRs that remove some of the `{{#view}}` helper usage and didn't want to potentially create merge conflicts against those PRs.

**Comment by rwjblue** [comment]
@chadhietala - I was suggesting that we want to keep the transform that asserts when using `{{view 'foo'}}` around until ~ 3.0.0. For example, in 2.0.0 we removed the transform that asserted on `{{each foo in bar}}` usage, and since that silently does absolutely nothing was a fairly large troll factor...

**Comment by chadhietala** [comment]
@rwjblue Yea that makes sense. What your thoughts about the coverage? Is it ok if we just delete the tests and not the `view` helper implementation?

**Comment by rwjblue** [comment]
The deletion definitely needs to happen, but I'd rather keep queue up the deletions and rebase (or merge master in here) as we go to land it all at once.

**Comment by chadhietala** [comment]
So lets land the test removals and then I'll open another PR that removes the helper, but it will needed to be shepherded until all the `view` helpers are removed from the tests.

**Comment by rwjblue** [comment]
These files have been deleted, closing...

**Pull request closed without merging**